### PR TITLE
Fix machineInfo curl validation

### LINKED ISSUE #456: Parsing logic for EnvDetector failed to grab CURL information
For example: version prints out: “curl 7.71.1 (powerpc-ibm-aix6.1.9.0) libcurl/7.71.1 OpenSSL/1.0.2u zlib/1.2.11 libssh2/1.8.2 nghttp2/1.41.0” The parser reads 1.0.2 instead of 7.71.1

### PR DESCRIPTION
**Pull request by renfeiw**
- update curl version command
- switch machineInfo detection to warning mode
- processBuilder to use env copied from the parent shell environment

Fixes #456

### CODE REVIEW

**Comment by renfeiw** [comment]
grinder: Grinder/33867/

File: src/org/openj9/envInfo/MachineInfo.java [hunk 1]
```diff
@@ -175,12 +176,11 @@ private void validateInfo() {
...
 		for (Info info : infoMap.values()) {
 			if (info.req != null) {
-				String version = parseInfo(info.output);
-				valid &= validateVersion(info.name, version, info.req);
+				valid &= validateVersion(info, info.req);
 			}
 		}
 		if (!valid) {
-			System.exit(1);
+//			System.exit(1);
```
Comment (karianna) [inline src/org/openj9/envInfo/MachineInfo.java:183]: remove commented out code?

**Review by karianna — CHANGES_REQUESTED** [review]

File: src/org/openj9/envInfo/MachineInfo.java (hunk 1 above)
Comment (renfeiw) [reply src/org/openj9/envInfo/MachineInfo.java]: This is on purpose. As described in the commit msg, we want to switch envDetector to warning mode. We will later remove the comment and add an option to toggle the failure or warning mode. I have added an comment on the code, so it is clearer.

**Review by karianna — APPROVED** [review]

**Review by llxia — APPROVED** [review]
LGTM

**Pull request merged**